src/components/top_bar/top_bar.ts

```diff
@@ -270,22 +285,84 @@ css/* scss */ `
         user-select: text;
       }
     }
+    .o-style-thin {
+      border-bottom: 1px solid #000000;
+      margin-top: 8px;
+      margin-bottom: 6px;
+      width: 60px;
+    }
+    .o-style-medium {
+      border-bottom: 2px solid #000000;
+      margin-top: 8px;
+      margin-bottom: 6px;
+      width: 60px;
+    }
+    .o-style-thick {
+      border-bottom: 3px solid #000000;
+      margin-top: 8px;
+      margin-bottom: 6px;
+      width: 60px;
+    }
+    .o-style-dashed {
+      border-bottom: 1px dashed #000000;
+      margin-top: 8px;
+      margin-bottom: 6px;
+      width: 60px;
+    }
+    .o-style-dotted {
+      border-bottom: 1px dotted #000000;
+      margin-top: 8px;
+      margin-bottom: 6px;
+      width: 60px;
+    }
+    .o-dropdown-border-type {
+      display: flex;
+    }
+    .o-dropdown-border-check {
+      width: 20px;
+      font-size: 16px;
+    }
   }
 `;
+
+const BORDER_POSITIONS = [
+  [
+    ["all", "o-spreadsheet-Icon.BORDERS"],
+    ["hv", "o-spreadsheet-Icon.BORDER_HV"],
+    ["h", "o-spreadsheet-Icon.BORDER_H"],
+    ["v", "o-spreadsheet-Icon.BORDER_V"],
+    ["external", "o-spreadsheet-Icon.BORDER_EXTERNAL"],
+  ],
+  [
+    ["left", "o-spreadsheet-Icon.BORDER_LEFT"],
+    ["top", "o-spreadsheet-Icon.BORDER_TOP"],
+    ["right", "o-spreadsheet-Icon.BORDER_RIGHT"],
+    ["bottom", "o-spreadsheet-Icon.BORDER_BOTTOM"],
+    ["clear", "o-spreadsheet-Icon.BORDER_CLEAR"],
+  ],
+];
+
 export class TopBar extends Component<Props, SpreadsheetChildEnv> {
   static template = "o-spreadsheet-TopBar";
   DEFAULT_FONT_SIZE = DEFAULT_FONT_SIZE;
+  BORDER_POSITIONS = BORDER_POSITIONS;
 
   static components = { ColorPicker, Menu, Composer };
   commonFormats = FORMATS;
   customFormats = CUSTOM_FORMATS;
   currentFormatName = "automatic";
   fontSizes = fontSizes;
 
+  borderStyles = ["thin", "medium", "thick", "dashed", "dotted"];
+
   style: Style = {};
   state: State = useState({
     menuState: { isOpen: false, position: null, menuItems: [] },
     activeTool: "",
+    subActiveTool: "",
+    borderStyle: DEFAULT_BORDER_DESC[0],
+    borderColor: DEFAULT_BORDER_DESC[1],
```

Comment:
The top bar should not be concerned about tools details such as border color or thickness.
You can create a subcomponent